src/background/services/openPayments.ts

```diff
@@ -947,11 +996,15 @@ export class OpenPaymentsService {
     return null;
   }
 
-  async rotateToken() {
+  async rotateToken(
+    options: { dedupeRejected: boolean } = { dedupeRejected: true },
+  ) {
     if (!this.grant) {
       throw new Error('No grant to rotate token for');
     }
-    const rotate = this.deduplicator.dedupe(this.client!.token.rotate);
+    const rotate = this.deduplicator.dedupe(this.client!.token.rotate, {
+      dedupeRejected: options.dedupeRejected,
+    });
```

Comment:
In the wallet reconnect flow, when the first `validateRedirect` rejects, it is cached for `5000` ms. This translates to the second `validateRedirect` attempt after adding the key, to return the rejected promise to the cache instead of trying again with a new request.

So I do not want to cache the token rotate when it comes to redirect, to I can try again to rotate the token at reconnect.

Reply:
@raducristianpopa thinking you're right.
This applies wherever  a rejected `rotateToken` is  followed by another `rotateToken` withing the `wait` cache clear.

`pay`, `payContinuous`, `adjustAmount` can use cached rejected promises instead of retrying to rotate.
Especially since I see that the `pay` website can call itself recursively if the token has expired.

cc @sidvishnoi

Reply:
Sounds right to me. Should still do a local manual test.

Aside: might be good to rename it to `cacheRejections`?